File: app/src/main/java/ptml/releasing/db/models/response/base/BaseResponse.java
package ptml.releasing.db.models.response.base;

import com.google.gson.annotations.SerializedName;
import org.jetbrains.annotations.NotNull;

public class BaseResponse {
    @SerializedName("message")
    private String message;

    @SerializedName("success")
    private boolean success;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public boolean isSuccess() {
        return success;
    }

    public void setSuccess(boolean success) {
        this.success = success;
    }


    public BaseResponse() {
    }

    public BaseResponse(String message, boolean success) {
        this.message = message;
        this.success = success;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof BaseResponse)) return false;

        BaseResponse that = (BaseResponse) o;

        if (success != that.success) return false;
        return message != null ? message.equals(that.message) : that.message == null;
    }

    @Override
    public int hashCode() {
        int result = message != null ? message.hashCode() : 0;
        result = 31 * result + (success ? 1 : 0);
        return result;
    }

    @NotNull
    @Override
    public String toString() {
        return "BaseResponse{" +
                "message='" + message + '\'' +
                ", success=" + success +
                '}';
    }
}

File: form/src/main/java/ptml/releasing/form/models/base/BaseResponse.kt
package ptml.releasing.form.models.base

import com.google.gson.annotations.SerializedName

open class BaseResponse {
    @SerializedName("message")
    var message: String? = null

    @SerializedName("success")
    var isSuccess: Boolean = false


    //used by serializers and deserializers
    constructor()

    constructor(message: String, success: Boolean) {
        this.message = message
        this.isSuccess = success
    }


    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is BaseResponse) return false

        if (message != other.message) return false
        if (isSuccess != other.isSuccess) return false

        return true
    }

    override fun hashCode(): Int {
        var result = message?.hashCode() ?: 0
        result = 31 * result + isSuccess.hashCode()
        return result
    }

    override fun toString(): String {
        return "BaseResponse(message=$message, isSuccess=$isSuccess)"
    }
}
